sql/srv_session.cc

```diff
@@ -1095,6 +1113,10 @@ int Srv_session::execute_command(enum enum_server_command command,
 
   mysql_audit_release(&thd);
 
+#ifdef WITH_WSREP
+  wsrep_before_command(&thd);
```

Comment:
If we look at "normal" flow, in sql_parse.cc do_command() which is caller of dispatch_command() (so is equivalent of this function) we call wsrep_before_command() as well, but we have more sophisticated logic around this which detects BF aborted transactions done by background rollbacker.
There is also logic that blocks queries in case of snapshot was not installed.

Don't we need this logic here as well?

Reply:
I followed the async slave (rpl) implementation for these calls, which doesn't implement the retry logic at this level.  It's probably too complicated to add at this point.